tasks/FileUploadTask.md

```diff
@@ -1 +1,153 @@
 # File Upload Task
+
+## Objectives
+
+Uploading a large file can be done by creating an upload session which allows the upload of chunks of the file in sequential API requests. This therefore enables the possibility mechanisms such as resuming the upload or cancelling the upload.
+This task aims to provide a fluent and easy to use mechanism for the consumer to control, monitor and cancel the handling of large uploads.
+
+## Requirements
+
+- Accept a Microsoft.Graph.UploadSession instance to be used for the upload which is created and managed by the user and handed over to the task. The task should use the instance to start/resume the upload.
```

Comment:
We should validate the expiration and byte range on resume by [querying the upload URL](https://docs.microsoft.com/en-us/graph/api/driveitem-createuploadsession?view=graph-rest-1.0#resuming-an-in-progress-upload).

Reply:
I've added this to the spec